util-core/src/main/scala/com/twitter/util/Batcher.scala

```diff
@@ -0,0 +1,20 @@
+package com.twitter.util
+
+/** Provides a clean, lightweight interface for working with batched Futures */
+private[util] class Batcher[In, Out](
+  executor: BatchExecutor[In, Out]
+)(
+  implicit timer: Timer
+) extends Function1[In, Future[Out]] { batcher =>
+  /** Enqueues requests for a batched Future */
```

Comment:
More I'm saying that if we wrap it in double square brackets then scaladoc will render it as an internal link.  It's not a huge deal, but it can be nice.  I'm OK with merging as-is.